primitives/src/vouches.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2023 Encointer Association
+// This file is part of Encointer
+//
+// Encointer is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Encointer is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Encointer.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::common::BoundedIpfsCid;
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::RuntimeDebug;
+use scale_info::TypeInfo;
+#[cfg(feature = "serde_derive")]
+use serde::{Deserialize, Serialize};
+
+#[derive(Default, Encode, Decode, Copy, Clone, PartialEq, Eq, Debug, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "serde_derive", derive(Serialize, Deserialize))]
+pub enum PresenceType {
+	#[default]
+	Unspecified,
+	Virtual,
+	Physical,
+}
+
+#[derive(Default, Encode, Decode, Copy, Clone, PartialEq, Eq, Debug, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "serde_derive", derive(Serialize, Deserialize))]
+#[repr(u8)]
+pub enum VouchType {
+	#[default]
+	Unspecified = 0,
+	KnownHuman = 10,
+	EncounteredHuman(PresenceType) = 20,
+	EncounteredObject(PresenceType) = 30,
+	VisitedEvent(PresenceType) = 40,
+	VisitedPlace(PresenceType) = 50,
```

Comment:
What do you want to achieve with the enum values and the `repr(u8)`? For backwards compatibility, we would always want to append at the end of the enum.

Reply:
I wanted a way that gives us backward compatibility while still allowing to maintain meaningful order. So I left gaps to leave space for more differentiation

Reply:
Hmm, I think then we should do it more in line with the scale-codec api: it is better to add:

```rust
 #[codec(index = 10)]
KnownHuman,
```

Reply:
why? then we still need `repr(u8)`
it just doubles the number of lines. what's the benefit? Doesn't it do the same?

Reply:
okok, clippy seems to agree with you

Reply:
actually, I think hard-coding for ordering doesn't actually work if we don't know what future variants might be. It might be necessary to change existing order for it to make any sense. dropped it